python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -306,6 +310,30 @@ def __init__(
                     "or be a star() expression."
                 )
 
+    def _detect_and_get_compute_strategy(
+        self, exprs: list["Expr"]
+    ) -> Optional[ComputeStrategy]:
+        """Detect if expressions contain callable class UDFs and return appropriate compute strategy.
+
+        If any expression contains a callable class UDF, returns ActorPoolStrategy.
+        Otherwise returns None (which will default to TaskPoolStrategy).
+        """
+        from ray.data._internal.planner.plan_expression.expression_visitors import (
+            _CallableClassUDFCollector,
+        )
+
+        # Check all expressions for callable class UDFs
+        for expr in exprs:
+            collector = _CallableClassUDFCollector()
+            collector.visit(expr)
+            if collector.get_callable_class_udfs():
+                # Found at least one callable class UDF - use actor semantics
+                from ray.data._internal.compute import ActorPoolStrategy
+
+                return ActorPoolStrategy(min_size=1, max_size=None)
```

Comment:
for regular UDFs, when using the actor pool strategy, now it's mandatory to specify concurrency. 
I think we should probably preserve this behavior and change it later for both regular and express-based UDFs.